### PR TITLE
Feature: declarative pods support in uberjars

### PR DESCRIPTION
**Pull request by cap10morgan**
Please answer the following questions and leave the below in as part of your PR.

- [X] I have read the [developer documentation](https://github.com/babashka/babashka/blob/master/doc/dev.md).

- [ ] This PR corresponds to an [issue with a clear problem statement](https://github.com/babashka/babashka/blob/master/doc/dev.md#start-with-an-issue-before-writing-code). - technically no. want me to write one? or it could be considered a part of #863 

- [ ] This PR contains a [test](https://github.com/babashka/babashka/blob/master/doc/dev.md#tests) to prevent against future regressions - not sure how testable this is...

- [ ] I have updated the [CHANGELOG.md](https://github.com/babashka/babashka/blob/master/CHANGELOG.md) file with a description of the addressed issue. - does this warrant its own changelog entry?

### CODE REVIEW

File: src/babashka/main.clj [hunk 1]
```diff
@@ -920,11 +920,15 @@ Use bb run --help to show this help output.
             (spit uberscript-out expression :append true)))
         (when uberjar
           (if-let [cp (cp/get-classpath)]
-            (uberjar/run {:dest uberjar
-                          :jar :uber
-                          :classpath cp
-                          :main-class main
-                          :verbose debug})
+            (do
+              (fs/copy (:file @common/bb-edn) "resources/bb.edn")
```
Comment (borkdude) [inline src/babashka/main.clj:924]: Is the assumption here that "resources" is always on the classpath?

File: src/babashka/main.clj [hunk 2]
```diff
@@ -939,17 +943,19 @@ Use bb run --help to show this help output.
...
         config (:config global-opts)
-        bb-edn-file (or config
-                        "bb.edn")
-        bb-edn (when (fs/exists? bb-edn-file)
-                 (System/setProperty "babashka.config"
-                                     (.getAbsolutePath (io/file bb-edn-file)))
+        abs-path #(-> % io/file .getAbsolutePath)
+        bb-edn-file (or (and config (fs/exists? config) (abs-path config))
+                        (some-> jar cp/loader (cp/resource "bb.edn") .toString)
+                        (and (fs/exists? "bb.edn") (abs-path "bb.edn")))
```
Comment (borkdude) [inline src/babashka/main.clj:951]: Won't this change load a bb.edn file although --config says it's another file, when that other file doesn't exist?

Comment (borkdude) [reply src/babashka/main.clj:951]: Maybe we should only load bb.edn from the uberjar when bb is executing that uberjar?

Comment (cap10morgan) [reply src/babashka/main.clj:951]: `(some-> jar ...)` does that, I think? If it isn't executing a jar, `jar` will be nil and `some->` will bail right there and the `or` will move on to `./bb.edn`. Let me know if you think I've gotten anything wrong there, though.

**Comment by borkdude** [comment]
I guess this test needs to be skipped on Windows: `lein test :only babashka.uberjar-test/uberjar-with-pods-test`

Can you give some background/explanation of why the arg parsing code had to change (with the risk of introducing other behavior)?

**Pull request merged**